app/main.py

```diff
@@ -1,8 +1,34 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        self.name = name
+        self.age = age
+
+        Person.people[self.name] = self
+
+
+def find_partner(person_obj: Person,
+                 person_to_find: str | None
+                 ) -> Person | str | None:
+    if person_to_find in Person.people:
+        partner = person_obj.people[person_to_find]
+        if hasattr(partner, "wife"):
+            partner.wife = person_obj
+        elif hasattr(partner, "husband"):
+            partner.husband = person_obj
+        return Person.people[person_to_find]
+
+
+def create_person_list(people: [dict]) -> [Person]:
+
+    for human_data in people:
+        curr_person = Person(human_data["name"], human_data["age"])
+        if human_data.get("wife"):
+            curr_person.wife = find_partner(curr_person, human_data["wife"])
+        elif human_data.get("husband"):
+            curr_person.husband = find_partner(curr_person,
+                                               human_data["husband"])
+
+    return list(Person.people.values())
```

Comment:
The function should return a list of objects. It returns objects. Please, can you tell me more about the problem, right now - I can't see a reason to change the code


Ofc it will return you different lists if you call it twice, is It a problem? I can't see "this" specified as a "must" in the description

Reply:
I can change it, but I want to know the reason

Reply:
no, I can't, due to tests I can't add a new variable "people_list"